docs/source/architecture/poet.rst

```diff
@@ -0,0 +1,595 @@
+  SGX SDK ```sgx_get_key()`` function passing a fixed 32 byte constant as
+  ``key_id`` argument.
+
+PoET\_MRENCLAVE
+  Public MRENCLAVE (see SGX SDK documentation) value of valid
+  PoET SGX enclave.
+
+:math:`T_{WT}`
+  WaitTimer timeout in seconds. A validator has at most :math:`T_{WT}`
+  seconds to consume a WaitTimer, namely obtain a WaitCertificate on it after
+  the WaitTimer itself has expired.
+
+:math:`K`
+  Number of blocks a validator can commit before having to sign-up with
+  a fresh PPK.
+
+:math:`c`
+  The "sign-up delay", i.e., number of blocks a validator has to wait after
+  sign-up before starting to participate in elections.
+
+minDuration
+  Minimum duration for a WaitTimer.
+
+2 - P2P PoET SGX Enclave Specifications
+=======================================
+The P2P PoET SGX enclave uses the following data structures::
+
+  WaitTimer {
+    double requestTime
+    double duration
+    byte[32] WaitCertId:sub:`n`
+    double localMean
+  }
+
+  WaitCertificate {
+    WaitTimer waitTimer
+    byte[32] nonce
+    byte[] blockDigest
+  }
+
+It uses the following global variables::
+
+  WaitTimer activeWT # The unique active WaitTimer object
+  byte[64] PPK
+  byte[64] PSK
+  MCID # SGX Monotonic Counter Identifier
+
+It exports the following functions:
+
+``generateSignUpData(OPKhash)``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Returns**
+
+.. code:: console
+
+    byte[64]  PPK
+    byte[432] report # SGX Report Data Structure
+    byte[256] PSEmanifest
+    byte[672] sealedSignUpData # (PPK, PSK, MCID) tuple encrypted with SealKey
+
+****Parameters****
+
+.. code:: console
+
+    byte[32] OPKhash # SHA256 digest of OPK
+
+**Description**
+
+1. Generate fresh ECC keypair (PPK, PSK)
+#. Create monotonic counter and save its identifier as MCID.
+#. Use SGX ``sgx_seal_data()`` function to encrypt (PPK, PSK, MCID) with
+   SealKey (using MRENCLAVE policy)
+   :math:`sealedSignupData = \textnormal{AES-GCM}_{SealKey} (PPK | PSK | MCID)`
+#. Create SGX enclave report, store ``SHA256(OPKhash|PPK)`` in ``report_data``
+   field.
+#. Get SGX PSE manifest: PSEManifest.
+#. Save (PPK, PSK, MCID) as global variables within the enclave.
+#. Set active WaitTimer instance activeWT to NULL.
+#. Return (PPK, report, PSEmanifest, sealedSignUpData).
+
+.. note::
+   **Implementation Note:** Normally there is a maximum number of monotonic
+   counters that can be ceteated. One way to deal with this limitation is to
+   destroy a previously created monotonic counter if this is not the first time
+   the generateSignupData function was called.
+
+``unsealSignUpData(sealedSignUpData)``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Returns**
+
+.. code:: console
+
+    byte[64] PPK
+
+**Parameters**
+
+.. code:: console
+
+    byte[672] sealedSignUpData # (PPK, PSK, MCID) tuple encrypted with SealKey
+
+**Description**
+
+1. Use ``sgx_unseal_data()`` function to decrypt sealedSignUpData into (PPK,
```

Comment:
It would be good to review all this type of text and add articles "Use _the_ function", etc.